### ISSUE TITLE
Duplicate when use init

### ISSUE THREAD

**Issue by xsktno1** [pinecone-router/router#38]
I had the problem that the warning alert(1) would repeat twice. Is there a way to only initialize once? 

const routes = [ {path: '/article/:slug', template: '/pages/article.html', handlers: []} ]

document.addEventListener('alpine:init', () => {
    try {
        window.PineconeRouter.settings.hash = false;
        window.PineconeRouter.settings.basePath = '/';
        window.PineconeRouter.settings.templateTargetId = 'app';
        routes.forEach(route => {
            PineconeRouter.add(route.path, {handlers: route.handlers, template: route.template});
        });
    } catch (error) {
        console.error('An error occurred:', error);
    }
});

pages/article.html

<div x-data="{
                data: {},
                slug: $router.params.slug,
                init: function () {
	            alert(1)
		}
            }"
>
</div>

Thank.

**Comment by rehhouari** [comment]
hey thanks for the report!
I'm gonna look into it but till I fix it, does using [x-handler](/pinecone-router/router#x-handler) suffice as a workaroud? since i'm sure it's called only once when the route is visited

**Comment by xsktno1** [comment]
I also tried x-hander but it's not stable. Hope you can support fetching data from init. Thanks.

```
async function articleHandler(ctx) {
    try {
        let response = await axios.get(API_URL+$ctx.params.slug);
        Alpine.store('data').article = response.data;
    } catch (error) {
        // window.PineconeRouter.context.navigate('/404')
    }
}
```
pages/article.html
<div x-data="{
                data: {},
                article: Alpine.store('data').article
            }"
>
</div>

**Comment by rehhouari** [comment]
Hi I fixed the duplicate init issue, you can try it out in version 4.3.2

**Comment by rehhouari** [comment]
Feel free to re-open if issue still persists or make another issue if there is another problem!

**Comment by xsktno1** [comment]
After updating to 4.3.2, I noticed the alert fix appearing twice. However, when I tried to fetch, I encountered the issue of fetching twice.

```
<div
	x-data='{
            data: {
                title: "",
                description: ""
            },
            init: async function () {
                try {
				let res = await fetch("https://abc.com/wp-json/wp/v2/posts")
						.then(response => response.json())
						.then(posts => {
						console.log(posts);
				})
				.catch(error => console.error("Error:", error));
                    this.setSettings();
                } catch (error) {
                    console.error(error);
                }
            },
```

<img width="952" alt="Screenshot 2024-04-16 at 20 28 14" src="https://github.com/pinecone-router/router/assets/37059033/ccc429b8-a205-4e80-8cff-f91dac82fddd">

**Comment by rehhouari** [comment]
I'll look into that too, I cant seem to get it the code you mentioned to work I just get `Alpine Expression Error: Unexpected token ')'` error

Also, can you mention what's not stable about handlers? cause they seem like they'd be good for this task.

**Comment by xsktno1** [comment]
Thank you for your kind attention to my issue.
